### PR TITLE
Custom krb5.conf file location

### LINKED ISSUE #86: Add support for custom krb5.conf
In https://exasol.atlassian.net/browse/L3-1417, client faced with requirement to specify their own krb5.conf, which we currently doesn't support. 

Here we should implement the support of this, as this might be needed in complex setups.
It will allow for user to specify the path to the file in the bucketfs and setup java configuration settings accordingly

Need to implement

- [x] parameter for the user to set the file path
- [x] check that file indeed exists
- [x] set configuration options to take the file into account
- [x] implement integration tests for this
- [x] update documentation

### PR DESCRIPTION
**Pull request by Shmuma**
Closes #86 

- [x] Parameter to be set with config file location
- [x] Check for file existence if parameter is given
- [x] Set configuration option to take it into account
- [x] unit tests for such use-case
- [x] documentation update

### CODE REVIEW

File: doc/changes/changes_1.7.3.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Exasol Kafka Connector Extension 1.7.3, released 2024-02-15
```
Comment (ckunki) [inline doc/changes/changes_1.7.3.md:1]: I am sorry, due to slow review you will need to update this before release.

File: doc/changes/changes_1.7.3.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Code name:
```
Comment (ckunki) [inline doc/changes/changes_1.7.3.md:3]: Please add a code name before releasing.

File: doc/changes/changes_1.7.3.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Exasol Kafka Connector Extension 1.7.3, released 2024-02-15
+
+Code name: 
+
+## Summary
+Implemented support for custom `krb5.conf` files.
```
Comment (ckunki) [inline doc/changes/changes_1.7.3.md:6]:
```suggestion

Implemented support for custom `krb5.conf` files.
```

File: doc/user_guide/user_guide.md [hunk 4]
```diff
@@ -489,6 +489,9 @@ keyTab="/buckets/bfsdefault/bucket1/kafka.keytab"
 principal="principal@DOMAIN.COM";
 ```
 
+In some complex setups, you might need to provide custom ``krb5.conf`` file. Thes could be done by uploading it to the 
+BucketFS and providing the path in ``SASL_KRB5CONF_LOCATION`` parameter, similar to ``SASL_JAAS_LOCATION``.
```
Comment (ckunki) [inline doc/user_guide/user_guide.md:493]:
```suggestion
In some complex setups, you might need to provide a custom ``krb5.conf`` file. Thes could be done by uploading it to the BucketFS and providing the path in ``SASL_KRB5CONF_LOCATION`` parameter, similar to ``SASL_JAAS_LOCATION``.
```

File: doc/user_guide/user_guide.md [hunk 5]
```diff
@@ -690,6 +693,9 @@ not in import statement itself.
   more complex configuration of SASL authentication. It should refer to the file
   stored inside a bucket in Exasol BucketFS.
 
+* ``SASL_KRB5CONF_LOCATION`` - It is the location of the custom ``krb5.conf`` file. 
+  It should refer to the file stored inside a bucket in Exasol BucketFS.
```
Comment (ckunki) [inline doc/user_guide/user_guide.md:697]:
```suggestion
* ``SASL_KRB5CONF_LOCATION`` - It is the location of the custom ``krb5.conf`` file. It should refer to the file stored inside a bucket in Exasol BucketFS. 
```

Comment (ckunki) [reply doc/user_guide/user_guide.md:697]: As (at least I) am always wondering about the paths in BucketFS, i.e. where to start and whether there must be a slash prefix, etc.: Could you give an example?

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 6]
```diff
@@ -299,6 +299,28 @@ class KafkaConsumerProperties(private val properties: Map[String, String]) exten
...
+    val krb5confLocation = get(SASL_KRB5CONF_LOCATION.userPropertyName)
+    if (krb5confLocation.isEmpty) {
+      ""
+    } else {
+      if (!Files.isRegularFile(Paths.get(krb5confLocation.get))) {
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-28")
+            .message("Unable to find the custom krb5.conf file at {{JAAS_LOCATION}}", krb5confLocation)
+            .mitigation(BUCKETFS_CHECK_MITIGATION)
```
Comment (ckunki) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:316]: Beware: Java compiler will not allow using constants in mitigations etc. 
But for scala (luckily or unluckily ?) there is no compiler :slightly_smiling_face:

**Review by ckunki — CHANGES_REQUESTED** [review]

File: doc/changes/changes_1.7.3.md (hunk 1 above)
Comment (Shmuma) [reply doc/changes/changes_1.7.3.md:1]: No problem. Let's aim to next Monday :). Will update

**Review by ckunki — APPROVED** [review]

**Pull request merged**